Update the logging page for 2025.01 (#2063)
neo4j/docs-operations: modules/ROOT/pages/monitoring/logging.adoc

| _debug.log_
| Log information required by Neo4j Customer Support to investigate problems.

> It is highly recommended to keep it enabled and not to alter the _debug.log_ format to ensure the best support.

Author: I see. I'll remove it.

> It is highly recommended to keep it enabled and not to alter the _debug.log_ format to ensure the best support.

Author: ```suggestion
It is highly recommended to keep it enabled and not to alter the _debug.log_ format.
```

Neo4j uses https://logging.apache.org/log4j/2.x/[Log4j 2] for logging.
The logging configuration is located in the _conf_ directory and consists of two files:

* _user-logs.xml_ -- provides configuration for _neo4j.log_.
* _server-logs.xml_ -- provides configuration for _debug.log_, _http.log_, _query.log_, and _security.log_.

The _gc.log_ is handled by the Java Virtual Machine(JVM) and is configured using the JVM parameters.
For details, see <<garbage-collection-log-file-configuration>>.

[json-format]
[NOTE]
====
Starting from Noe4j 2025.01, the default configuration files are located in the _server.directories.configuration=conf_ directory.

> Therefore, the default paths to the configuration files are _conf/user-logs.xml_ and _conf/server-logs.xml_.

Author: ```suggestion
Therefore, the default paths to the configuration files are _<home>/conf/user-logs.xml_ and _<home>/conf/server-logs.xml_.
```

> And <conf> by default is <home>/conf.

Author: ```suggestion
```

> If you use a custom non-absolute paths set in the _neo4j.conf_ file for either of your Log4j configuration files, by using the parameters xref:configuration/configuration-settings.adoc#config_server.logs.user.config[`server.logs.user.config`] and xref:configuration/configuration-settings.adoc#config_server.logs.config[`server.logs.config`], then these values will likely be wrong, as these settings now depend on <conf> rather than <home>.

Author: How about:
```suggestion
If you use a custom directory for your Log4j configuration files, update the paths to your XML files in the _neo4j.conf_ file to depend on <conf> rater than <home>. 
For example, xref:configuration/configuration-settings.adoc#config_server.logs.user.config[`server.logs.user.config=path/to/your/conf/user-logs.xml`] and xref:configuration/configuration-settings.adoc#config_server.logs.config[`server.logs.config=path/to/your/conf/server-logs.xml`].
```

<4> `filePattern` specifies a file pattern to be used when the file is rolled.
The pattern renames the files to _debug.log.01_ and _http.log.01_ when they reach the defined trigger.

<5> The `JsonTemplateLayout` layout is used to format the log messages in JSON format.
Starting from Neo4j 2025.01, the default _debug.log_ format is JSON, and it is recommended to keep it enabled to ensure the best support.

> If you require the _debug.log_ messages in a different format, create an additional Appender.

Reviewer: maybe here we should say that we recommend to create separate appended since default one is used by neo4j support.
Also maybe we should notice that the previous layout is there and commented out so to make migration back simpler if required?

> If you require the _debug.log_ messages in a different format, create an additional Appender.

Author: ```suggestion
If you require the _debug.log_ messages in a different format, create an additional Appender because the default one is used by Neo4j support.
If you require to use the plain format, you can uncomment entry <6>.
```

> <6> `PatternLayout` defines the layout for the appender, in this case with the `GMT+2` timezone.

Reviewer: maybe a note that its a disabled one actually and this is the old one pre json format?

Reviewer: I think only this one is left

Author: Ah, I might have missed it.

<4> `filePattern` specifies a file pattern to be used when the file is rolled.
The pattern renames the files to _debug.log.01_ and _http.log.01_ when they reach the defined trigger.

<5> The `JsonTemplateLayout` layout is used to format the log messages in JSON format.
Starting from Neo4j 2025.01, the default _debug.log_ format is JSON, and it is recommended to keep it enabled to ensure the best support.
If you require the _debug.log_ messages in a different format, create an additional Appender since the default one is used by the Neo4j Support team.
If you require to use the text format, you can uncomment entry <6>.

> <6> `PatternLayout` defines the layout for the appender, in this case with the `GMT+2` timezone.

Author: ```suggestion
<6> `PatternLayout` defines the layout for the appender, in this case, with the `GMT+2` timezone.
Note that this is used in the 5.x series and is now disabled by default.
```
